clang/lib/CIR/CodeGen/CIRGenBuilder.cpp

```diff
@@ -95,15 +95,17 @@ void CIRGenBuilderTy::computeGlobalViewIndicesFromFlatOffset(
       unsigned AlignMask = Layout.getABITypeAlign(Elts[I]).value() - 1;
       if (RecordTy.getPacked())
         AlignMask = 0;
-      Pos = (Pos + AlignMask) & ~AlignMask;
+      if (!RecordTy.isUnion())
```

Comment:
Can you add some comment here so it's easy to understand why unions are different?